legal-api/src/legal_api/reports/report.py

```diff
@@ -312,6 +312,9 @@ def _set_dates(self, filing):
                 filing['effective_date'] = agm_date.strftime('%B %-d, %Y')
             else:
                 filing['agm_date'] = 'No AGM'
+                # When AR indicates NO AGM the effective date of the filing is Dec 31st of the AR year
+                ar_year_str = str(filing.get('annualReport', {}).get('annualReportDate', datetime).year)
+                filing['effective_date'] = 'December 31, ' + ar_year_str
```

Comment:
I think we should update effective_date in db instead of just showing in report. @lekshmimallika-aot @argush3 thoughts?

Reply:
yes, i think so too.

Reply:
Okay I will make that change.

Reply:
Yes, Needs a db update